interfaces/cython/cantera/yaml2ck.py

```diff
@@ -0,0 +1,1029 @@
+            reaction_equation = reaction.equation
+
+        return f"{reaction_equation:<{self.max_rxn_eqn_width}}"
+
+    def _reorder_reaction_equation(self, reaction):
+        """
+        Reorders reactions based on species molecular weights
+        """
+        solution = self.solution
+
+        # Split Reaction Equation
+        rxn_eqn = reaction.equation
+        for reaction_direction in [" <=> ", " <= ", " => "]:
+            if reaction_direction in rxn_eqn:
+                break
+        for third_body in [" (+M)", " + M", ""]:  # search eqn for third body
+            if third_body in rxn_eqn:  # if reaches '', doesn't exist
+                break
+
+        # Sort and apply to reaction equation
+        reaction_txt = []
+        reaction_split = {"reactants": reaction.reactants, "products": reaction.products}
+        for species in reaction_split.values():
+            species_weights = []
+            for key in species.keys():
+                index = solution.species_index(key)
+                species_weights.append(solution.molecular_weights[index])
+
+            # Append coefficient to species
+            species_list = []
+            for species_text, coef in species.items():
+                if coef == 1.0:
+                    species_list.append(species_text)
+                else:
+                    species_list.append(f"{coef:.0f} {species_text}")
+
+            species = species_list
+
+            # Reorder species based on molecular weights
+            species = [x for y, x in sorted(zip(species_weights, species))][::-1]
+            reaction_txt.append(" + ".join(species) + third_body)
+
+        reaction_txt = reaction_direction.join(reaction_txt)
+
+        return reaction_txt
+
+    def _build_coeffs(self, coeffs):
+        coeffs = [self.eformat(coef) for coef in coeffs]
+        return ((" ")*self.coef_space).join(coeffs)
+
+    def _build_arrhenius(self, rate, rxn_order):
+        """Builds Arrhenius coefficient string based on reaction type.
+        Parameters
+        ----------
+        rate : ct.Arrhenius
+            Arrhenius-form reaction rate coefficient
+        rxn_order : int or float
+            Order of reaction
+        Returns
+        -------
+        str
+            String with Arrhenius coefficients
+        """
+
+        A  = rate.pre_exponential_factor*1e3**(rxn_order)
+        N  = rate.temperature_exponent
+        Ea = rate.activation_energy / CALORIES_CONSTANT
+
+        return self._build_coeffs([A, N, Ea])
+
+    def _build_line_1_text(self, reaction, rxn_note=None):
+        """
+        Builds the first line including the reaction equation, basic Arrhenius
+        coefficients, and the after-line note
+        """
+        rxn_eqn = self._build_reaction_equation(reaction)
+
+        # The Arrhenius parameters that follow the equation string on the main line
+        # depend on the type of reaction.
+        if type(reaction) is ct.ElementaryReaction:
```

Comment:
With the newest `main`, you’ll see that this reaction type is deprecated (several others below are as well) and won’t even be instantiated if you import a mechanism from YAML, meaning that this will fail with the latest 2.6.0a4 version. Instead, you’ll have to test for the type of the reaction rate.